esp/esp/program/models/class_.py

```diff
@@ -1111,7 +1115,27 @@ def firstBlockEvent(self):
     def isFull(self, ignore_changes=False, webapp=False):
         if len(self.get_meeting_times()) == 0:
             return True
-        if webapp and Tag.getBooleanTag('count_checked_in_only', program = self.parent_program, default = False):
+
+        now = datetime.datetime.now()
+        switch_time = None
+        if Tag.getProgramTag('switch_time_program_attendance', program=self.parent_program):
+            try:
+                switch_hour, switch_min = [int(x) for x in Tag.getProgramTag('switch_time_program_attendance', program=self.parent_program).split(":")]
+                switch_time = datetime.datetime(now.year, now.month, now.day, switch_hour, switch_min)
+            # TODO: Is this the best way to deal with someone specifying the tag incorrectly?
+            except ValueError:
+                pass
+        switch_lag = None
+        if Tag.getProgramTag('switch_lag_class_attendance', program = self.parent_program):
+            try:
+                switch_lag = int(Tag.getProgramTag('switch_lag_class_attendance', program = self.parent_program))
+            except ValueError:
+                pass
+        # TODO: Don't call num_students_attending() twice, but also don't call it unless needed?
+        if webapp and switch_lag and now >= (self.start_time_prefetchable() + timedelta(minutes=switch_lag)) and self.num_students_attending() >= 1:
+            num_students = self.num_students_attending()
```

Comment:
Hmm...yeah, that's tricky. I can't think of a way to avoid the two calls here, but we could probably cache `num_students_attending()` above so that it's only one database call maximum. `self.students()` might already have some caching going on, too, so if that's the case, you wouldn't even need to worry about it.